src/typechecker/FStar.TypeChecker.Rel.fst

```diff
@@ -1788,6 +1787,205 @@ let simplify_guard env g = match g.guard_f with
       in
       {g with guard_f=f}
 
+//
+// Apply substitutive indexed effects subcomp for an effect M
+//
+// bs: (opened) binders in the subcomp type
+// subcomp_c: the computation type in the subcomp type (opened with bs)
+// ct1 ct2: the two input computation types, both in M
+// sub_prob: a function to create and add subproblems to the worklist
+// num_effect_params: number of effect parameters in M
+// wl: worklist
+// subcomp_name and r1: for debugging purposes
+//
+// returns the (subcomp guard, new sub problems, worklist)
+//
+let apply_substitutive_indexed_subcomp (env:Env.env)
```

Comment:
Applying a substitutive indexed effect subcomp. It mainly computes substitutions for the combinator binders.